Add SSM2211

zoonman opened the pull request:
Datasheet:
https://www.analog.com/media/en/technical-documentation/data-sheets/SSM2211.pdf

I am not completely sure that I added filters for footprints because the part could come in 2 different kinds of packages. But both have the same pinout. Please, help me with it.

There is an EPAD pin in LFSCP which supposed to be connected to `V-`.

<img width="1091" alt="Screen Shot 2019-08-10 at 23 45 08" src="https://user-images.githubusercontent.com/576297/62829336-2da32d00-bbc9-11e9-82f2-93195f628e98.png">

Symbol:

<img width="988" alt="Screen Shot 2019-08-10 at 23 38 05" src="https://user-images.githubusercontent.com/576297/62829268-ef593e00-bbc7-11e9-95f6-dd25e5925b87.png">

chschlue commented:
Shutdown is active-high, why did you call it ~SD?

LFCSP-8-1EP_3x3mm_P0.5mm_EP1.45x1.74mm should be the right one.

HTH

zoonman commented:
@chschlue I might be wrong but I assembled this schematic and I did work.

<img width="552" alt="Screen Shot 2019-08-11 at 09 21 41" src="https://user-images.githubusercontent.com/576297/62834409-c82b5c80-bc19-11e9-801c-7f3290f6cf70.png">

I think you are right, to activate shutdown there should be logical 1 on `SD` pin. 
I will redraw schematic. 
But I still not completely sure about package. 
How can I specify 2 different packages and configure EPAD properly. Travis doesn't seem to like it.

chschlue commented:
EP pin number should be 9, name V-. Stacking is fine.
Default FP should be empty.
FP filters should be SOIC?8\*3.9x4.9mm\*P1.27mm\* and LFCSP\*1EP\*3x3mm\*P0.5mm\*

zoonman commented:
@chschlue is there a policy for merging or something?
How do you decide when symbols are going to be merged?

poeschlr commented:
We merge as soon as we are sure everything is ok. This of course depends on a librarian finding time to review a contribution. (This can take a while as we are short on manpower. If you want to help out simply review a few pull requests for correctness and you might just get an invitation mail to be officially part of the team.)

poeschlr commented:
And on the last point: @chschlue you seem interested (you made quite some reviews lately). Would you like to officially join the team and gain merge privilege?

chschlue commented:
@poeschlr Sure, if it helps.
But note that I actually haven't reviewed anything thoroughly yet, IIRC. I usually just commented on what I happened to notice.

poeschlr commented:
@chschlue this is how everyone of us started out. We made a few reviews, got noticed and invited to the team. Some then gradually take on more and more responsibility, others help out for a while and then move on. 
I just sent you an invitation for that reason. You are free to ignore it if you do not want to take on the slight increase in responsibility (after all you can merge as soon as you accept.) It is also not expected that you know everything from the get go. 
Just remember that you always can ask us experienced guys if anything is unclear.

chschlue commented:
@zoonman Change pin 1 to input and call it SHUTDOWN to reflect the datasheet name, then this should be good to go.
Perhaps switch pins 1 and 2 so the symbol doesn't get unnecessarily large and possibly 3 and 4 while you're at it to ease wiring for typical applications.

zoonman commented:
@chschlue I made requested updates. I hope everybody will be happy with these changes.

<img width="677" alt="Screen Shot 2019-10-07 at 20 19 25" src="https://user-images.githubusercontent.com/576297/66358195-e2df3080-e93f-11e9-9b2f-b8d9076a8f89.png">

evanshultz commented:
I would like to see a triangle-shaped symbol used. While there is a slight crash of text and outline, here is a suitable symbol we can up with for differential input and output parts like this:
![image](https://user-images.githubusercontent.com/1936989/66435711-546bbd00-e9da-11e9-9ce9-53e34ded571d.png)

From the branch diff and the screenshot above, I have some comments:
1. The text crashing on pins 2 and 7 is quite ugly. More space, please.
1. The refdes and value are floating above the top of the symbol.
1. Because the CSP package needs a pin 9, you cannot have both footprints sharing one symbol. Pick a footprint for your one symbol or make two different symbols.
1. Once you've done the above, put the package (such as `LFCSP-8`) at the end of the description.
1. If you go with the SOIC package, do not include the pin count in the FP filter. Remove the `?8` part.

Also, pin 1 should be Input type.

chschlue commented:
@zoonman ping

zoonman commented:
@chschlue I will finalize this PR soon. Sorry, it got off my radar.

zoonman commented:
@chschlue @evanshultz I updated symbol <img width="1027" alt="Screen Shot 2019-12-16 at 00 10 28" src="https://user-images.githubusercontent.com/576297/70880858-909e3980-1f98-11ea-87c2-5e15dcd1ffa5.png">

To be totally honest I found the current symbols implementation flawed and having too much hassle about things that doesn't make any sense to me. Like I said I am not a hardware engineer, thus brings me to a point where I consider components totally interchangeable as long as they suffice the product requirements. 
For the current component it is obvious that both of versions of the component are totally identical except the package. 
My opinion - every component should have a special "Model" property which holds exact model name (see Ordering Guide in the manual) and can be tied to exact package option. This field also could help to deal with BOM generation and future integrations with PCB Houses. Also introducing "virtually connected" mount pins could help with LFCSP packages.

So far I am planning to go with SSM2211SZ as separate symbol because it has 9 pins and incompatible with SSM2211. I'll prefer to have a second PR just to get this stuff merged.

TBH, the whole contribution experience is negative. It is bureaucratic. Everyone hates 90 degrees turns when everything seems almost done. Requirements should be clear, consistent and documented. Also, symbol checkers should be part of Symbol Editor so I don't have to contribute symbols when those aren't ready. It would be much nicer to have "contribute" button in Symbol Editor which will take people to a proper page with contribution guidelines etc. It is easy to implement and it might help you in library components growth. Even I could implement such button for you.

I know I have negative outlook at the whole thing but there are ways for improvement. KiCad is a great software with an immersive potential. Just don't be afraid to experiment a little bit and introduce new things. 

I have certain ideas but those might appear very unconventional for u so I slightly afraid to share them.

evanshultz commented:
This library cannot fulfill all purposes to everyone. While it is the official KiCad library, no claim is made for perfection or inclusiveness. But thanks to the community, we try. Maybe nobody is happy. You obviously aren't delighted. I hope many are.

Full, orderable MPNs are out-of-scope because from who I order may not be from who you order. Or my preferred supplier may not be available where you live. There are KiCad libraries available from some parts vendors. And there are KiCad extensions which can help with managing MPNs.

If by "virtually connected" pins you mean one symbol pin that goes to multiple footprint pads, KiCad already supports this. Implementations can always be questioned but it is possible.

You are correct that an 8-pin and 9-pin packages for the "same" part need unique symbols.

What I see happened above is that you contributed a symbol, then got a peer review, then that peer made more suggestions immediately after becoming a librarian, and then I reviewed the symbol. We worked as a team of volunteers and I do not know what you mean by "90 degree turns". Most of the suggestions above are complementary.

The library requirements are at http://kicad-pcb.org/libraries/klc/. Not everything has a guideline because that's too onerous, and some areas of the library are old and non-conformant, but the library requirements are fairly-well documented and improving all the time.

There are both symbol and footprint checkers available to run offline, and they are part of the CI run on this and all other PRs. They are not part of the KiCad UI because everyone can have different rules.

Many of your comments should be directed at the KiCad software and not the library, which you will find at https://gitlab.com/kicad/code/kicad. They are separate entities.

zoonman commented:
@evanshultz I understood. I know it is difficult to find time.

So, shall we merge this pull request? I think, I took into account all your notes.

zoonman commented:
@evanshultz @chschlue are we on track to merge?

zoonman commented:
@evanshultz @chschlue I updated the library one more time. 
I double checked symbol names and updated them to correspond with ordering information to avoid possible confusion between packages.
I used stacked pins for LFCSP package according to the manual. 
The _only_ thing that Travis is not happy about - I used 40mils name for shutdown pin BUT it is OK according to the KLC. So we must be good.

chschlue commented:
- [ ] Leave out _Z_ suffix indicating RoHS compliance

Looks good to me otherwise.

zoonman commented:
I don't see non-RoHS parts. I think it is better to keep it with Z.
<img width="1188" alt="Screen Shot 2019-12-28 at 16 34 58" src="https://user-images.githubusercontent.com/576297/71549720-29b55300-2990-11ea-9f0e-1395bdc2a958.png">

zoonman commented:
@chschlue @evanshultz ???

evanshultz commented:
Thanks for taking the time to review and respond to the comments by librarians. Note that commenting more is unlikely to give us more free time to review your submission.

https://kicad-pcb.org/libraries/klc/S2.2/ is clear that RoHS compliance is one of the wildcarded, non-functional parts of a MPN. As you were already told above. So remove the 'Z'.

More things for you to fix:
1. In general the description format I prefer is `datasheet description[, unique information], package`. But this isn't codified anywhere and what you have written captures the relevant details.
1. Change `4Mhz` in the description to `4MHz`.
1. The end of each description should be `LFCSP-8` or `SOIC-8`. Just that. Not two package sections.
1. The CPZ symbol needs to have `1EP*` added after the package type in the FP filter to make sure a thermal pad-equipped footprint is matched.

Lastly, please give a relevant commit message to each commit. Above it appears you used the default commit message of `Update Amplifier_Audio.[dcm|lib]` which is not helpful to reviewers.

chschlue commented on Amplifier_Audio.dcm:
```diff
@@ -270,6 +270,18 @@ K audio microphone amplifier
+$ENDCMP
+#
+$CMP SSM2211S
+D 1.5W BTL or 250mW SE 4MHz-bandwidth high performance amplifier, SOIC_N
```
Change to `…, SOIC-8`

chschlue commented:
Thanks.

poeschlr commented:
@chschlue tip for the future: You can edit the commit message when using squash merge. Githubs default suggestion is to include all commit message headers of the squashed commit range. This leads to strange results if the user did not really care about their messages like here. Right now this commit was created with >10 lines of the same repeated message.

I generally change it to say add "device type" "part number". Only thing you really need to keep is the (#pr-number) part as this will allow us to later find the pull request easily.
